js/src/Lines.ts

```diff
@@ -592,74 +591,76 @@ export class Lines extends Mark {
+    compute_view_padding(): void {
         //This function sets the padding for the view through the variables
-        //x_padding and y_padding which are view specific paddings in pixel
-        let x_padding;
+        //xPadding and yPadding which are view specific paddings in pixel
+        let xPadding;
         if (this.model.get("marker")) {
-            const marker_padding = Math.sqrt(this.model.get("marker_size")) / 2 + 1.0;
-            const line_padding = this.model.get("stroke_width") / 2.0;
-            x_padding = Math.max(marker_padding, line_padding);
+            const markerPadding = Math.sqrt(this.model.get("marker_size")) / 2 + 1.0;
+            const linePadding = this.model.get("stroke_width") / 2.0;
+            xPadding = Math.max(markerPadding, linePadding);
         } else {
-            x_padding = this.model.get("stroke_width") / 2.0;
+            xPadding = this.model.get("stroke_width") / 2.0;
         }
 
-        const y_padding = x_padding;
-        if(x_padding !== this.x_padding || y_padding !== this.y_padding) {
-            this.x_padding = x_padding;
-            this.y_padding = y_padding;
+        const yPadding = xPadding;
+        if(xPadding !== this.xPadding || yPadding !== this.yPadding) {
+            this.xPadding = xPadding;
+            this.yPadding = yPadding;
             this.trigger("mark_padding_updated");
         }
     }
 
-    update_marker_style() {
-        const that = this;
-        const fill_color = this.model.get("fill_colors");
+    update_marker_style(): void {
+        const fillColor = this.model.get("fill_colors");
         const opacities = this.model.get("opacities");
-        this.d3el.selectAll(".curve").each(function(d, i) {
+        this.d3el.selectAll(".curve").each((d, i) => {
             const curve = d3.select(this);
             curve.selectAll(".dot")
                 .style("opacity", opacities[i])
-                .style("fill", that.get_mark_color(d, i) || fill_color[i]);
+                .style("fill", this.get_mark_color(d, i) || fillColor[i]);
         });
     }
 
-    update_marker(model, marker) {
+    //removed unused var model
+    update_marker(marker): void {
         if (marker) {
             this.draw_dots();
             this.update_dots_xy(false);
             this.update_marker_style();
-            if (this.legend_el) {
-                this.legend_el.select(".dot").attr("d", this.dot.type(marker).size(25));
+            if (this.legendEl) {
+                this.legendEl.select(".dot").attr("d", this.dot.type(marker).size(25));
             }
         } else {
             this.d3el.selectAll(".dot").remove();
-            if (this.legend_el) {
-                this.legend_el.select(".dot").attr("d", this.dot.size(0));
+            if (this.legendEl) {
+                this.legendEl.select(".dot").attr("d", this.dot.size(0));
             }
         }
     }
 
-    update_marker_size(model, marker_size) {
+    //removed unused var model
```

Comment:
Nope, that will not work :) This method is used by `this.listenTo(this.model, "change:marker_size", this.update_marker_size);` and passes the `model` as first argument. I suppose this is not tested well.

Reply:
Should we create tests for these?

Reply:
Yes, more test coverage is good!

You will see your code fail if you try to change the `marker_size` attribute from Python.

Reply:
You can't really change the signature of a function/method and expect it to work as intended, even if you don't use some of those arguments.

It's not super-well documented in Backbone, but I remember the callback function passed to [listenTo](https://backbonejs.org/#Events-listenTo) is expected to have a signature like this: `function(model, value): void`, `model` being the object you listen to, and `value` I suppose is the value of the attribute you listen to (Don't take my words as granted, the documentation being what it is I think backbone's source code might be the only source of truth).

So if you change the signature like you are doing, the parameter `marker_size` will actually contain `this.model` instead of the marker size, that's why it doesn't work.

Reply:
huh, I just wrote the following test:

```

def test_marker_size(scales):
    lines = bqplot.Lines(x=[[0, 1], [0, 1, 2]], y=[[0, 1], [1, 0, -1]], scales=scales)

    lines.marker_size = 7
```

Using the code without the `model` variable and the test passed.
Did I write the test wrong?

Reply:
That doesn't work, unfortunately. You are only testing the Python code here, the is no actual JavaScript model and view that are generated. The Python will run just fine, it's the JavaScript code that will fail.

You can try in a Notebook and you should see an error in the browser console.

Reply:
But you can actually add a JavaScript test directly here: https://github.com/bqplot/bqplot/blob/master/js/src/test/lines.ts, and changing the `marker_size` doing something like: `model.set('marker_size', 7)`

Reply:
ah, right :)
thanks Martin!
I actually didn't find the js tests thought we only had the python ones :see_no_evil: